docs/conf.py

```diff
@@ -361,6 +361,9 @@ def custom_defined_func():
 # build needs.json to make permalinks work
 needs_build_json = True
 
+# build json file include needs with the same docs_name
+needs_per_page = True
```

Comment:
This option name should be consistent with the other builders (`build_json`, `build_json_per_id`), i.e. `build_json_per_page`

Thinking holistically, with there now being numerous data writers, and more on the way, maybe it makes sense to have some form of unified configuration option.

For example:

```python
needs_data_writers = ["json", "json_per_id", "json_per_page"]
```